packages/rum-core/src/performance-monitoring/transaction-service.js

```diff
@@ -352,6 +355,43 @@ class TransactionService {
     )
   }
 
+  setSession(tr) {
+    let session = this._config.get('session')
+    if (session) {
+      if (typeof session == 'boolean') {
+        tr.session = {
+          id: generateRandomId(16),
+          sequence: 1
+        }
+      } else {
+        if (
+          session.timestamp &&
+          Date.now() - session.timestamp > SESSION_TIMEOUT
+        ) {
+          tr.session = {
+            id: generateRandomId(16),
+            sequence: 1
+          }
+        }
+
+        tr.session = {
+          id: session.id,
+          sequence: session.sequence ? session.sequence + 1 : 1
```

Comment:
This feels wrong, what does sequence refer to when it comes to measuring session length? 

Right now, this would be increasing for each and every `click` transactions on the page which feels incorrect. Is this what we want?

Reply:
Actually, we currently use sessions only for page loads (and maybe route-changes in the future), so click transactions are not involved. 

Please see the original issue for a bit more explanation on the sequence field, but happy to go into more detail if needed.